File: RecipeFrontend/recipe-book/src/app/models/recipe.ts
import { RecipeTag } from "./recipe-tag";

export class Recipe{
  constructor(
    recipeId: number,
    title: string,
    description: string,
    cookingTime: number,
    portions: number,
    image: string,
    recipeFavouritesCount: number,
    recipeLikesCount: number,
    recipeTags: Array<RecipeTag>,
  ){
    this.recipeId = recipeId
    this.title = title
    this.description = description
    this.cookingTime = cookingTime
    this.portions = portions
    this.image = image
    this.recipeFavouritesCount = recipeFavouritesCount
    this.recipeLikesCount = recipeLikesCount
    this.recipeTags = recipeTags
  }
  
  recipeId: number;
  title: string;
  description: string;
  cookingTime: number;
  portions: number;
  image: string;
  recipeFavouritesCount: number;
  recipeLikesCount: number;
  recipeTags: Array<RecipeTag>;
}

File: RecipeBook/RecipeBook/Application/Models/Recipe.cs
namespace RecipeBook.Application.Models
{
    public class Recipe
    {
        public Recipe()
        {
        }
        public Recipe(
            string title,
            string description,
            int cookingTime,
            int portions,
            string image,
            ICollection<RecipeIngredientsBlock> recipeIngredientsBlocks,
            ICollection<RecipeStep> recipeSteps,
            ICollection<RecipeTag> recipeTags,
            int recipeLikesCount,
            int recipeFavouritesCount)
        {
            Title = title;
            Description = description;
            CookingTime = cookingTime;
            Portions = portions;
            Image = image;
            CreatingDate = DateTime.Today;
            RecipeIngredientsBlocks = recipeIngredientsBlocks;
            RecipeSteps = recipeSteps;
            RecipeLikesCount = recipeLikesCount;
            RecipeFavouritesCount = recipeFavouritesCount;
            RecipeTags = recipeTags;
        }

        public int RecipeId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int CookingTime { get; set; }
        public int Portions { get; set; }
        public string Image { get; set; }
        public DateTime CreatingDate { get; set; }
        public ICollection<RecipeIngredientsBlock> RecipeIngredientsBlocks { get; set; }
        public ICollection<RecipeStep> RecipeSteps { get; set; }
        public ICollection<RecipeTag> RecipeTags { get; set; }
        public int RecipeLikesCount { get; set; }
        public int RecipeFavouritesCount { get; set; }
    }
}
